File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/context/DataTableContext_ViewTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "DataTableContext_ViewModelTestEnvironment.h"
#include "DataTableContext_ViewModel.h"
#include "DataTableContext_ViewModelTestEnvironmentImpl.h"
#include <string>

namespace context
{
  class DataTableContext_ViewTests : public testing::Test
  {
  public:
    std::shared_ptr<DataTableContext_ViewModel> sut;
    std::shared_ptr<DataTableContext_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    std::string dataTableCustomSetter = R"(| id | name |
| 0 | A |)";
    virtual void given_dataTableCustomSetter();
    std::string dataTableToString = R"(| id | name |
| 0 | A |)";
    virtual void given_dataTableToString();
    std::string dataTableToJson = R"([ { id:"0", name:"A" } ])";
    virtual void given_dataTableToJson();
    std::string dataTableToXml = R"(<Data>
  <DataRow>
    <id>0    </id>
    <name>A    </name>
| 0 | A |
  </DataRow>
</Data>)";
    virtual void given_dataTableToXml();
  protected:
    void SetUp() override;
  };
  void DataTableContext_ViewTests::SetUp()
  {
    this->testEnvironment = std::make_shared<DataTableContext_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  TEST_F(DataTableContext_ViewTests, DataTable_Context_Custom_Setter_given_dataTableCustomSetter_when_then_)
  {
    this->given_dataTableCustomSetter();
    this->BuildSut();
  }
  TEST_F(DataTableContext_ViewTests, DataTable_Context_to_String_given_dataTableToString_when_then_)
  {
    this->given_dataTableToString();
    this->BuildSut();
  }
  TEST_F(DataTableContext_ViewTests, DataTable_Context_to_JSON_given_dataTableToJson_when_then_)
  {
    this->given_dataTableToJson();
    this->BuildSut();
  }
  TEST_F(DataTableContext_ViewTests, DataTable_Context_to_XML_given_dataTableToXml_when_then_)
  {
    this->given_dataTableToXml();
    this->BuildSut();
  }
  void DataTableContext_ViewTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void DataTableContext_ViewTests::given_dataTableCustomSetter()
  {
    this->testEnvironment->SetIdAndName(this->dataTableCustomSetter);
  }
  void DataTableContext_ViewTests::given_dataTableToString()
  {
    this->testEnvironment->SetDataTableString(this->dataTableToString);
  }
  void DataTableContext_ViewTests::given_dataTableToJson()
  {
    this->testEnvironment->SetDataTableJson(this->dataTableToJson);
  }
  void DataTableContext_ViewTests::given_dataTableToXml()
  {
    this->testEnvironment->SetDataTableXml(this->dataTableToXml);
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/context/DataTableContext_ViewTests.cs
/// <filename>
///     DataTableContext_ViewTests.cs
/// </filename>
using context;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class DataTableContext_ViewTests
{
    private context.DataTableContext_ViewModel sut;

    private context.DataTableContext_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new context.DataTableContext_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void DataTable_Context_Custom_Setter_given_dataTableCustomSetter_when_then_()
    {
        this.given_dataTableCustomSetter();
        this.BuildSut();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void DataTable_Context_to_String_given_dataTableToString_when_then_()
    {
        this.given_dataTableToString();
        this.BuildSut();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void DataTable_Context_to_JSON_given_dataTableToJson_when_then_()
    {
        this.given_dataTableToJson();
        this.BuildSut();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void DataTable_Context_to_XML_given_dataTableToXml_when_then_()
    {
        this.given_dataTableToXml();
        this.BuildSut();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    private string dataTableCustomSetter = @"| id | name |
| 0 | A |";

    public virtual void given_dataTableCustomSetter()
    {
        this.testEnvironment.SetIdAndName(this.dataTableCustomSetter);
    }

    private string dataTableToString = @"| id | name |
| 0 | A |";

    public virtual void given_dataTableToString()
    {
        this.testEnvironment.SetDataTableString(this.dataTableToString);
    }

    private string dataTableToJson = @"[ { id:""0"", name:""A"" } ]";

    public virtual void given_dataTableToJson()
    {
        this.testEnvironment.SetDataTableJson(this.dataTableToJson);
    }

    private string dataTableToXml = @"<Data>
  <DataRow>
    <id>0    </id>
    <name>A    </name>
| 0 | A |
  </DataRow>
</Data>";

    public virtual void given_dataTableToXml()
    {
        this.testEnvironment.SetDataTableXml(this.dataTableToXml);
    }
}
